src/analyze.ts

```diff
@@ -130,3 +133,11 @@ export function findExports (code: string): ESMExport[] {
     return !nextExport || exp.type !== nextExport.type || !exp.name || exp.name !== nextExport.name
   })
 }
+
+function filterCommentCode (code: string) {
+  const matchedComments = matchAll(COMMENT_RE, code, { type: 'comment' })
+  for (const matched of matchedComments) {
+    code = code.replace(matched.code, '')
```

Comment:
Can we please run regex to replace once with `replaceAll`? (We can fallback to `replace` for Node < 15 support)

Reply:
Using replacAll is indeed a good option, but does it require the user node version >= 15, do we have to limit the user?

Reply:
@pi0 we can also use the code to determine which method to use
```ts
function filterCommentCode (code: string) {
  if (typeof String.prototype.replaceAll === 'function') {
    code = code.replaceAll(COMMENT_RE, '').trim()
  } else {
    const matchedComments = matchAll(COMMENT_RE, code, { type: 'comment' })
    for (const matched of matchedComments) {
      code = code.replace(matched.code, '').trim()
    }
  }
  return code
}
```

Reply:
Nice solution for fallback. But I think we don't need replaceAll when `/g` is there neither trimming. Can you please check it?